src/epjson/epJSONTranslator.cpp

```diff
@@ -121,20 +132,32 @@ JSONValueType schemaPropertyTypeDecode(const Json::Value& type) {
   return JSONValueType::NumberOrString;
 }
 
-JSONValueType getSchemaObjectPropertyType(const Json::Value& schema, const std::string& type_description, const std::string& property_name) {
-  auto type = schemaPropertyTypeDecode(safeLookupValue(getSchemaObjectProperties(schema, type_description), property_name, "type"));
-  if (type == JSONValueType::NumberOrString) {
-    LOG_FREE(LogLevel::Warn, "epJSONTranslator",
-             "Unknown value passed to schemaPropertyTypeDecode, returning generic 'NumberOrString' Option. "
-               << "Occurred for property_name=" << property_name);
+/** Locate the properties for a given field of an object.
+ * - group_name can be empty, or it's the name of the extensible group.
+ * - field_name is the name of the field
+ * - field_property_name is something like 'type', 'anyOf', or 'enum'
+ *
+ * if group_name is empty:
+ *  getSchemaObjectProperties > [field_name] > [field_property_name]
+ * else:
+ *  getSchemaObjectProperties > [group_name] > items > properties > [field_name] > [field_property_name]
+ */
+const Json::Value& getSchemaObjectFieldProperty(const Json::Value& schema, const std::string& type_description, const std::string& group_name,
+                                                const std::string& field_name, const std::string& field_property_name) {
+
+  const auto& objectProperties = getSchemaObjectProperties(schema, type_description);
+
+  if (group_name.empty()) {
+    return safeLookupValue(objectProperties, field_name, field_property_name);
+  } else {
+    return safeLookupValue(objectProperties, group_name, "items", "properties", field_name, field_property_name);
   }
-  return type;
 }
 
-JSONValueType getSchemaObjectPropertyType(const Json::Value& schema, const std::string& type_description, const std::string& group_name,
-                                          const std::string& field_name) {
-  auto type = schemaPropertyTypeDecode(
-    safeLookupValue(getSchemaObjectProperties(schema, type_description), group_name, "items", "properties", field_name, "type"));
+/** Find the 'type' property of a field and convert that to enum.*/
+JSONValueType getSchemaObjectFieldPropertyType(const Json::Value& schema, const std::string& type_description, const std::string& group_name,
+                                               const std::string& field_name) {
+  JSONValueType type = schemaPropertyTypeDecode(getSchemaObjectFieldProperty(schema, type_description, group_name, field_name, "type"));
```

Comment:
Change name and use of functions a bit. One calls the other. (group_name will always be passed, even if empty)